File: samples/discovery/ios/ios/ComponentBoxUI/Sources/ComponentBoxUI/components/Text.swift
//
// Created by Matt Ramotar on 3/27/22.
// Copyright (c) 2022 Dropbox, Inc. All rights reserved.
//

import Foundation
import SwiftUI
import componentbox

@available(iOS 14.0, *)
struct ComponentBoxUIText: View, Identifiable {
    let id: String
    let modifier: componentbox.Modifier?
    let text: String?
    let color: componentbox.Color?
    let textStyle: String?
    
    func getColor() -> SwiftUI.Color {
        color != nil ? color!.title.ui() : textStyle == "button" ? Color.ui.onPrimary : Color.ui.onBackground
    }

    var body: some View {

        let isHeading = ["h1", "h2", "h3", "h4", "h5", "h6"].contains(self.textStyle)
        let isButton = ["button"].contains(self.textStyle)

        if (isHeading) {
            SwiftUI.Text(text ?? "").sharpGroteskBold(style: .title3, color: getColor())
        } else if (isButton) {
            SwiftUI.Text(text ?? "").sharpGroteskBold(style: .body, color: getColor())
        } else {
            SwiftUI.Text(text ?? "").atlasGrotesk(style: .body, weight: .regular, color: getColor() )
        }
    }
}

File: componentbox/src/commonMain/kotlin/com/dropbox/componentbox/component/Text.kt
package com.dropbox.componentbox.component

import com.dropbox.componentbox.foundation.Events
import com.dropbox.componentbox.foundation.Color
import com.dropbox.componentbox.foundation.Modifier
import com.dropbox.componentbox.foundation.TextStyle
import com.dropbox.componentbox.impl.ComponentBoxText

interface Text : Component {
    var text: String?
    val color: Color?
    val textStyle: TextStyle?

    companion object {
        fun of(
            modifier: Modifier? = null,
            events: Events? = null,
            text: String? = null,
            color: Color? = null,
            textStyle: TextStyle? = null,
        ): Text = ComponentBoxText(
            modifier = modifier,
            events = events,
            text = text,
            color = color,
            textStyle = textStyle,
        )
    }
}